From 7006d51816d255d8f8652d878b305d7c19fe4657 Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Fri, 10 Feb 2023 10:17:20 +0800
Subject: [PATCH] hse: pkcs: secboot: replace memcpy() with specific
 hse_memcpy()

When running hse-secboot command to enable secure boot feature
form user space, there is a "Bus error" reported. The reason is that
multiple bytes is copied into a share memory region that is defined as
device  memory, and their address don't align with even address.
For fixing this issue, NXP creates a specific memory copy function
hse_memcpy() to work around this issue. So, replace memcpy()
with specific hse_memcpy().

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 examples/hse-secboot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/hse-secboot.c b/examples/hse-secboot.c
index 0901664..7c8b7b7 100644
--- a/examples/hse-secboot.c
+++ b/examples/hse-secboot.c
@@ -352,7 +352,7 @@ int hse_smr_install(int fd, struct ivt *ivt, struct app_boot_hdr *app_boot)
 	smr_entry.versionOffset = 0;
 
 	/* workaround for bus error on writing to hse rmem */
-	memcpy(smr_entry_hse, &(smr_entry), sizeof(smr_entry));
+	hse_memcpy(smr_entry_hse, &(smr_entry), sizeof(smr_entry));
 
 	srv_desc.srvId = HSE_SRV_ID_SMR_ENTRY_INSTALL;
 	srv_desc.hseSrv.smrEntryInstallReq.accessMode = HSE_ACCESS_MODE_ONE_PASS;
-- 
2.36.1

